src/pages/home/sidebar/SidebarNavigationContext.js

```diff
@@ -0,0 +1,59 @@
+import React, {useMemo, useCallback, useState} from 'react';
+import PropTypes from 'prop-types';
+import CONST from '../../../CONST';
+import Navigation from '../../../libs/Navigation/Navigation';
+import GLOBAL_NAVIGATION_MAPPING from '../../../GLOBAL_NAVIGATION_MAPPING';
+
+const propTypes = {
+    /** Children to wrap. The part of app that should have acces to this context */
+    children: PropTypes.node.isRequired,
+};
+
+const SidebarNavigationContext = React.createContext({
+    selectedGlobalNavigationOption: undefined,
+    selectedSubNavigationOption: undefined,
+    updateFromNavigationState: () => {},
+});
+
+function mapSubNavigationOptionToGlobalNavigationOption(SubNavigationOption) {
+    // eslint-disable-next-line no-restricted-syntax
+    for (const [key] of Object.entries(GLOBAL_NAVIGATION_MAPPING)) {
+        if (GLOBAL_NAVIGATION_MAPPING[key].includes(SubNavigationOption)) {
+            return key;
+        }
+    }
+    return undefined;
+}
+
+function SidebarNavigationContextProvider({children}) {
+    const [selectedGlobalNavigationOption, setSelectedGlobalNavigationOption] = useState(CONST.GLOBAL_NAVIGATION_OPTION.CHATS);
+    const [selectedSubNavigationOption, setSelectedSubNavigationOption] = useState();
+
+    const updateFromNavigationState = useCallback(
+        (navigationState) => {
+            const topmostCentralPaneRouteName = Navigation.getTopMostCentralPaneRouteName(navigationState);
+            if (!topmostCentralPaneRouteName) {
+                return;
+            }
+
+            setSelectedSubNavigationOption(topmostCentralPaneRouteName);
+            setSelectedGlobalNavigationOption(mapSubNavigationOptionToGlobalNavigationOption(topmostCentralPaneRouteName));
+        },
+        [setSelectedGlobalNavigationOption, setSelectedSubNavigationOption],
+    );
+
+    const contextValue = useMemo(
+        () => ({
+            selectedGlobalNavigationOption,
+            selectedSubNavigationOption,
+            updateFromNavigationState,
+        }),
+        [selectedGlobalNavigationOption, selectedSubNavigationOption, updateFromNavigationState],
+    );
```

Comment:
It's a good practise to keep contextValue memoized.
I'm even adding eslint rule to enforce memoization - (currently still draft PR) https://github.com/Expensify/App/pull/28549
https://react.dev/reference/react/useContext#optimizing-re-renders-when-passing-objects-and-functions - a little description of the problem (if context value is not memoized).
> The component will re-render only if (...)

It can also be re-rendered by parent re-render